packages/common/src/api/signUp.ts

```diff
@@ -1,13 +1,111 @@
+import { isEmpty } from 'lodash'
+
 import { createApi } from 'audius-query'
+import { FeatureFlags } from 'services/remote-config/feature-flags'
+import { IntKeys } from 'services/remote-config/types'
+import { parseHandleReservedStatusFromSocial } from 'utils/handleReservedStatus'
+
+import { userApiFetch } from './user'
+
+const DEFAULT_HANDLE_VERIFICATION_TIMEOUT_MILLIS = 5_000
+
+const promiseWithTimeout = (promise: Promise<any>, timeout: number) => {
+  return Promise.race([
+    promise,
+    new Promise((resolve) => {
+      setTimeout(() => {
+        resolve(null)
+      }, timeout)
+    })
+  ])
+}
 
 const signUpApi = createApi({
   reducerPath: 'signUpApi',
   endpoints: {
     isEmailInUse: {
-      fetch: async ({ email }, { audiusBackend }) => {
+      fetch: async ({ email }: { email: string }, { audiusBackend }) => {
         return await audiusBackend.emailInUse(email)
       },
       options: {}
+    },
+    isHandleInUse: {
+      fetch: async ({ handle }: { handle: string }, context) => {
+        const user = await userApiFetch.getUserByHandle(
+          {
+            handle,
+            currentUserId: null,
+            retry: false
+          },
+          context
+        )
+        return !isEmpty(user)
+      },
+      options: {}
+    },
+    getHandleReservedStatus: {
+      fetch: async (
+        { handle }: { handle: string },
+        { audiusBackend, remoteConfigInstance, env }
+      ) => {
+        const handleCheckTimeout =
+          remoteConfigInstance.getRemoteVar(
+            IntKeys.HANDLE_VERIFICATION_TIMEOUT_MILLIS
+          ) ?? DEFAULT_HANDLE_VERIFICATION_TIMEOUT_MILLIS
+
+        if (env.ENVIRONMENT === 'production') {
+          const twitterCheckPromise = remoteConfigInstance.getFeatureEnabled(
+            FeatureFlags.VERIFY_HANDLE_WITH_TWITTER
+          )
+            ? promiseWithTimeout(
+                audiusBackend.twitterHandle(handle),
+                handleCheckTimeout
+              )
+            : null
+
+          const instagramCheckPromise = remoteConfigInstance.getFeatureEnabled(
+            FeatureFlags.VERIFY_HANDLE_WITH_INSTAGRAM
```

Comment:
Would these ever be false?

Reply:
not sure i just ported from old signup. @raymondjacobson do you know if these flags can be removed?

Reply:
we should leave this in as these tend to be kind of flakey flows / if instagrams blocks us for whatever reason we would want to turn them off

Reply:
we could move it to remote vars though since it is a boolean versus an experiment